### PR TITLE
PWN-4400 - Popular tokens + header viewholder added

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4400

## Description of Work

 Popular tokens + header viewholder added

## Screenshots

Android Emulator - Pixel_5_API_30:5554 2022-08-03 11-52-48

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/ui/main/empty/BigBannerViewHolder.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        when (val background: Drawable = viewBannerBackground.background) {
+            is ShapeDrawable -> {
+                background.paint.color = getColor(backgroundColorRes)
+            }
+            is GradientDrawable -> {
+                background.setColor(getColor(backgroundColorRes))
+            }
+            is ColorDrawable -> {
+                background.color = getColor(backgroundColorRes)
+            }
```
Comment (kuznetsov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/empty/BigBannerViewHolder.kt:44]: Why do you need this check if background is ShapeDrawable?

**Review by kuznetsov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/main/empty/BigBannerViewHolder.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/empty/BigBannerViewHolder.kt:44]: depending on background

File: app/src/main/res/layout/item_get_token.xml [hunk 2]
```diff
@@ -1,94 +1,81 @@
...
+            android:textColor="@color/night"
+            app:layout_constraintBottom_toTopOf="@id/textViewTotal"
+            app:layout_constraintEnd_toStartOf="@id/textViewValue"
+            app:layout_constraintStart_toEndOf="@id/imageViewToken"
+            app:layout_constraintTop_toTopOf="@id/imageViewToken"
+            tools:text="Solana" />
 
-        </androidx.constraintlayout.widget.ConstraintLayout>
+        <TextView
+            android:id="@+id/textViewTotal"
```
Comment (gslevinkov) [inline app/src/main/res/layout/item_get_token.xml:49]: textViewTotal -> textViewTokenRate

File: app/src/main/res/layout/item_get_token.xml [hunk 3]
```diff
@@ -1,94 +1,81 @@
...
+            app:layout_constraintBottom_toBottomOf="@id/imageViewToken"
+            app:layout_constraintEnd_toStartOf="@id/textViewValue"
+            app:layout_constraintStart_toEndOf="@id/imageViewToken"
+            app:layout_constraintTop_toBottomOf="@id/textViewName"
+            app:layout_constraintVertical_bias="1"
+            tools:text="23.8112 SOL" />
 
-    </androidx.cardview.widget.CardView>
+        <org.p2p.uikit.components.UiKitButton
+            android:id="@+id/textViewValue"
```
Comment (gslevinkov) [inline app/src/main/res/layout/item_get_token.xml:66]: buttonGetToken

File: app/src/main/res/layout/item_get_token.xml [hunk 4]
```diff
@@ -1,94 +1,81 @@
...
+            style="@style/UiKit.Components.Button.Small"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:backgroundTint="@color/rain"
```
Comment (gslevinkov) [inline app/src/main/res/layout/item_get_token.xml:71]:
```suggestion
            android:backgroundTint="@color/bg_rain"
```

File: app/src/main/res/layout/item_get_token.xml [hunk 5]
```diff
@@ -1,94 +1,81 @@
...
+            android:textAppearance="@style/WalletTheme.TextAppearance.Bold16"
+            android:textColor="@color/night"
```
Comment (gslevinkov) [inline app/src/main/res/layout/item_get_token.xml:73]:
```suggestion
            android:textColor="@color/text_night"
```

File: app/src/main/res/layout/item_get_token.xml [hunk 6]
```diff
@@ -1,94 +1,81 @@
...
 
-    </androidx.cardview.widget.CardView>
+        <org.p2p.uikit.components.UiKitButton
+            android:id="@+id/textViewValue"
+            style="@style/UiKit.Components.Button.Small"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:backgroundTint="@color/rain"
+            android:textAppearance="@style/WalletTheme.TextAppearance.Bold16"
```
Comment (gslevinkov) [inline app/src/main/res/layout/item_get_token.xml:72]: style should be from UiKit

File: app/src/main/res/layout/item_main_header.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/textViewHeader"
+    style="@style/UiKit.TextAppearance.SemiBold.Text1"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginHorizontal="16dp"
+    android:layout_marginVertical="8dp"
+    android:ellipsize="end"
+    android:maxLines="1"
+    android:textColor="@color/night"
```
Comment (gslevinkov) [inline app/src/main/res/layout/item_main_header.xml:13]:
```suggestion
    android:textColor="@color/text_night"
```

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomeFragment.kt [hunk 8]
```diff
@@ -150,10 +137,14 @@ class HomeFragment :
 
     private fun onFragmentResult(requestKey: String, result: Bundle) {
         result.getParcelable<Token>(KEY_RESULT_TOKEN)?.let {
-            replaceFragment(BuySolanaFragment.create(it))
+            buyToken(it)
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/HomeFragment.kt:140]: my fragment - can =))) uhahahah

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt [hunk 9]
```diff
@@ -105,7 +112,21 @@ class HomePresenter(
                     val isAccountEmpty = updatedTokens.run { size == 1 && first().isSOL && first().isZero }
                     when {
                         isAccountEmpty -> {
+                            val tokensForBuyOrReceive = userInteractor.getTokensForBuy(POPULAR_TOKENS.toList())
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:115]: no we do not have it!

File: app/src/main/java/org/p2p/wallet/home/ui/main/empty/EmptyViewAdapter.kt [hunk 10]
```diff
@@ -23,21 +22,29 @@ class EmptyViewAdapter(
...
 
     override fun getItemCount(): Int = data.size
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder = when (viewType) {
-        R.layout.item_big_banner -> BigBannerViewHolder(parent) { bannerId: Int -> listener.onBannerClicked(bannerId) }
-        R.layout.item_get_token -> GetTokenViewHolder(parent, listener)
+        R.layout.item_big_banner -> BigBannerViewHolder(parent) { bannerId: Int ->
+            listener.onBannerClicked(bannerId)
+        }
+        R.layout.item_get_token -> GetTokenViewHolder(parent) { popularToken ->
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/empty/EmptyViewAdapter.kt:36]: no GetTokenViewHolder or just PopularTokenViewHolder

File: app/src/main/java/org/p2p/wallet/home/ui/main/empty/GetTokenViewHolder.kt [hunk 11]
```diff
@@ -1,26 +1,70 @@
...
+            if (token.isRenBTC) {
+                R.string.main_popular_token_action_receive_button
+            } else {
+                R.string.main_popular_token_action_buy_button
+            }
+        )
+
+        val rate = if (token is Token.Active) {
+            token.usdRateOrZero.asUsd()
+        } else null
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/empty/GetTokenViewHolder.kt:51]: not working!

[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/home/ui/main/empty/GetTokenViewHolder.kt:51]: ah, yes, i forgot about type casting, sorry!

File: app/src/main/res/layout/item_main_header.xml [hunk 12]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<TextView
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/textViewHeader"
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply app/src/main/res/layout/item_main_header.xml:5]: i think we will reuse this title on main screen!

[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/res/layout/item_main_header.xml:5]: yeah, okay, let's try it

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**